platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py

```diff
@@ -267,12 +267,16 @@ def get_cpld_component_list(cls):
     @classmethod
     @utils.read_only_cache()
     def is_module_host_management_mode(cls):
+        sai_profile_file_syncd = os.path.join('/tmp/', 'sai.profile')
+        if os.path.exists(sai_profile_file_syncd):
```

Comment:
I would suggest to add an "else" - if this file doesn't exist for some reason, try to check the under-sku file. 
Need to think if we want to have this approach - 
on the one hand it may cause the bug to appear again. on the second hand, it is under /tmp/ folder and might be deleted. we want to continue the flow even if file was deleted from inside the docker.